<launch>
    <arg name="gui" value="true" />
    
    <node pkg="depthai_mapping" type="depthai_mapping_node" name="depthai_mapping_node" output="screen">
		<param name="mono_resolution" value="800p" />
        <param name="depth_map_topic" value="/depth" />

        <!-- DEPTHAI Config parameters -->
        <param name="confidence" value="225" />
        <param name="lr_check_thres" value="225" />
        <param name="median_kernal_size" value="7" />

        <param name="lr_check" value="true" />
        <param name="extended" value="true" />
	    <param name="subpixel" value="false" />

        <!-- local parameters -->
        <param name="max_range_clamp" value="10.0" />
        <param name="update_interval" value="0.1" />

        <param name="subsample" value="2" />
        <param name="downsample" value="1" />

        <!-- Statistical Outlier Removal Filter parameters -->
        <param name="std_dev_mul" value="0.10" />
        <param name="mean_k" value="100" />

        <!-- WLS Disparity filter parameters -->
        <param name="lambda" value="80.0" />
        <param name="sigma" value="1.0" />
    </node>

    
    <node name="my_rqt_dashboard" pkg="rqt_gui" type="rqt_gui" respawn="false" output="screen" args="--perspective-file $(find depthai_mapping)/depth_disparity.perspective" if="$(arg gui)"/>
    

</launch>
